### PR TITLE
dev-util/hip: fix a runtime bug and a QA issue

### PR DESCRIPTION
**Pull request by littlewu2508**
1. Adopt a patch from Debian to parse --hip-version, which fix a runtime bug. Apply on version 5.4.3 and 5.3.3.
2. Correct a patch bumping; fix a QA issue

### CODE REVIEW

File: dev-util/hip/hip-5.3.3-r4.ebuild [hunk 1]
```diff
@@ -97,11 +98,16 @@ src_prepare() {
...
+	mv bin/hipvars.pm "${T}"/hipvars.pm.orig
+	cp "$(prefixify_ro "${FILESDIR}"/hipvars.pm)" bin/hipvars.pm || die "failed to replace hipvars.pm"
+	sed -n '/^\$HIP_BASE_VERSION_.* = "/p' "${T}"/hipvars.pm.orig >> bin/hipvars.pm || die
+	cat <<-EOF >> bin/hipvars.pm || die
+	\$HIP_VERSION_MAJOR="\$HIP_BASE_VERSION_MAJOR";
+	\$HIP_VERSION_MINOR="\$HIP_BASE_VERSION_MINOR";
+	\$HIP_VERSION_PATCH="\$HIP_BASE_VERSION_PATCH";
+	\$HIP_VERSION="\$HIP_BASE_VERSION_MAJOR.\$HIP_BASE_VERSION_MINOR.\$HIP_BASE_VERSION_PATCH";
+	EOF
+	sed	-e "s,@CLANG_PATH@,${LLVM_PREFIX}/bin," -i bin/hipvars.pm || die
```
Comment (heroxbd) [inline dev-util/hip/hip-5.3.3-r4.ebuild:110]: This logic is too complex to figure out what you are trying to fix hipvars.pm.  Is it possible to simplify it?

Comment (littlewu2508) [reply dev-util/hip/hip-5.3.3-r4.ebuild:110]: Maybe I can put the lines appended in another file. But overall that does not simplify much.

Actually, I'd like to try the upstream `hipvars.pm`. Maybe after several versions of refining it become easy to adopt.

**Comment by heroxbd** [comment]
littlewu2508 ***@***.***> writes:

> +	sed -n '/^\$HIP_BASE_VERSION_.* = "/p' "${T}"/hipvars.pm.orig >> bin/hipvars.pm || die

use `grep`.

> +	sed	-e ***@***.***_PATH@,${LLVM_PREFIX}/bin," -i bin/hipvars.pm || die

replace '\t' with ' '.

**Comment by heroxbd** [comment]
littlewu2508 ***@***.***> writes:

> Actually, I'd like to try the upstream hipvars.pm. Maybe after several
> versions of refining it become easy to adopt.

If you want to try it.  Better try it now, the above intermediate state
is harder to maintain than both cases.

**Comment by littlewu2508** [comment]
> If you want to try it. Better try it now, the above intermediate state is harder to maintain than both cases.

It seems promising to use vanilla hipvars.pm, just needs a small patch to modify the `HIP_CLANG_PATH` and `CUDA_PATH`, then the output of `hipconfig` remains same compared to using the self-customized `hipvars.pm` (after c5b595e2e405cfcaa6bccae89600cd217f38ca5e). `hipcc` works identically, too.

**Comment by littlewu2508** [comment]
> It seems promising to use vanilla hipvars.pm, just needs a small patch to modify the `HIP_CLANG_PATH` and `CUDA_PATH`, then the output of `hipconfig` remains same compared to using the self-customized `hipvars.pm` (after [c5b595e](https://github.com/gentoo/gentoo/commit/c5b595e2e405cfcaa6bccae89600cd217f38ca5e)). `hipcc` works identically, too.

Well, do not merge now (79acd30a99a5dde9c6ad1445fd2f2f5ed905e6bb) -- I observed some test suite not built properly, if I managed to turn on tests (https://github.com/gentoo/gentoo/pull/29930/commits/cda527707ade57f718ceaee5897769d58cce9eee)

**Comment by littlewu2508** [comment]
> Well, do not merge now ([79acd30](https://github.com/gentoo/gentoo/commit/79acd30a99a5dde9c6ad1445fd2f2f5ed905e6bb)) -- I observed some test suite not built properly, if I managed to turn on tests ([cda5277](https://github.com/gentoo/gentoo/commit/cda527707ade57f718ceaee5897769d58cce9eee))

It's ready now -- I pushed a patch to fix FindHIP.cmake. There are changes needs to be adjust for testing but I'll leave that in another PR later. For this PR I think it's in good shape.

**Pull request closed without merging**